src/main/java/br/com/metricminer2/scm/GitRepository.java

```diff
@@ -410,4 +422,21 @@ public String blame(String file, String commitToBeBlamed, Integer line) {
 		}
 	}
 
+	public int getMaxNumberFilesInACommit() {
+		if(this.maxNumberFilesInACommit == null){
+			return DEFAULT_MAX_NUMBER_OF_FILES_IN_A_COMMIT;
+		} else {
+			return this.maxNumberFilesInACommit;
+		}
+	}
+
+	@Override
+	public void setMaxNumberFilesInACommit(int maxNumber) {
```

Comment:
Yes, you are right! So, initializing that field lazily (the way it is now) works!

For GitRemoteRepository, you can have a single static method in the GitRepository, asking for all the parameters, it doesn't need to be that fancy.